drivers/flash/nrf_qspi_nor.c

```diff
@@ -556,9 +587,29 @@ static int qspi_nor_write(struct device *dev, off_t addr, const void *src,
 		return -EINVAL;
 	}
 
-	qspi_lock(dev);
 
-	nrfx_err_t res = nrfx_qspi_write(src, size, addr);
+	nrfx_err_t res;
+#if IS_ENABLED(NORDIC_QSPI_NOR_FLASH_ALLOW_STACK_USAGE_FOR_DATA_IN_FLASH)
```

Comment:
Not sure this needs to be configurable now.

Reply:
As you commented below it has a performance penalty. This allows the user to either use this workaround or copy their data in RAM in a local function doing it more efficiently.

Reply:
You can use `IS_ENABLED` in C if condition, it will just evaluate to 0 or 1, compiler will throw away the branch that will never get executed so the result will be the same but code will look more clear. Just use it in `if` in line 593.

Reply:
Move  `IS_ENABLED` into if at line 593.

Reply:
Moved.

Reply:
> This allows the user to either use this workaround or copy their data in RAM in a local function doing it more efficiently.

They can still do that if there's no config. If they use a RAM buffer, this buffer will not be used.